Formula/apache-arrow.rb

```diff
@@ -15,7 +14,6 @@ class ApacheArrow < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "boost"
```

Comment:
Do we really don't need `boost` anymore?
It still exists in requirements https://arrow.apache.org/docs/developers/cpp/building.html#build-dependency-management

Reply:
These appear to be the cases where `boost` is needed: https://github.com/apache/arrow/blob/5673c43a20db86c89c80cc7f29c21ab3009019b5/cpp/cmake_modules/ThirdpartyToolchain.cmake#L707-L713

We should pass `-DBOOST_SOURCE=SYSTEM` to ensure that bundled/downloaded Boost is not used, even if we end up not needing to depend on Boost. Or perhaps even the global `-DARROW_DEPENDENCY_SOURCE=SYSTEM` which covers all dependencies.

Reply:
We've been working to make the arrow build lighter by default. Unless you're building with gcc 4.8, which I don't think you are, you no longer need boost to build arrow with the cmake flags enabled here.

Reply:
So, do we need to do anything for ensuring bundled boost is not in use or not?

Reply:
It won't use boost at all, bundled or otherwise, unless you're on gcc 4.8. Does Homebrew build bottles from source with gcc 4.8?

Reply:
Fair enough